test/artifact_test.js

```diff
@@ -235,6 +235,81 @@ suite('Artifacts', function() {
 
     });
 
+    test('S3 single part private artifact fetched with signed url', async () => {
+      let taskId = slugid.v4();
+      
+      debug('### Creating task');
+      await helper.queue.createTask(taskId, taskDef);
+
+      debug('### Claiming task');
+      await helper.queue.claimTask(taskId, 0, {
+        workerGroup:    'my-worker-group',
+        workerId:       'my-worker',
+      });
+
+      let uploadInfo = await client.prepareUpload({
+        filename: bigfilename,
+        forceSP: true,
+      });
+
+      let name = 'signed-url.dat';
```

Comment:
You'll want to make a name that uses `+`  and whitespace ` ` or whatever it was charaters that require encoding that was affected by previous commits in this PR.

Probably add a bunch of special characters :)